src/metabase/search/postgres/index.clj

```diff
@@ -122,9 +122,18 @@
                        [:setweight [:to_tsvector
                                     [:inline tsv-language]
                                     [:cast
-                                     (:searchable_text entity)
+                                     (:searchable_text entity "")
                                      :text]]
-                        [:inline "B"]]])))
+                        [:inline "B"]]]
+       :with_native_query_vector [:||
+                                  [:setweight [:to_tsvector [:inline tsv-language] [:cast (:name entity) :text]]
```

Comment:
Rather than creating this field independently, we could look at using a generated field which adds onto the other one. Will look into this when I make this all more generic.